tokio-macros/src/entry.rs

```diff
@@ -1,13 +1,139 @@
 use proc_macro::TokenStream;
+use proc_macro2::Span;
 use quote::quote;
-use std::num::NonZeroUsize;
+use syn::spanned::Spanned;
 
 #[derive(Clone, Copy, PartialEq)]
-enum Runtime {
-    Basic,
+enum RuntimeFlavor {
+    CurrentThread,
     Threaded,
 }
 
+impl RuntimeFlavor {
+    fn from_str(s: &str) -> Result<RuntimeFlavor, String> {
+        match s {
+            "current_thread" => Ok(RuntimeFlavor::CurrentThread),
+            "multi_thread" => Ok(RuntimeFlavor::Threaded),
+            "single_threaded" => Err("The single threaded runtime flavor is called `current_thread`.".to_string()),
```

Comment:
I think this has potential for confusion, we do have a single threaded variant?

Reply:
It was mentioned [a few times][1] in the issue. The idea was that the single-thread runtime is a current-thread runtime running on a thread controlled by the `Runtime` object. The reason we don't want this variant on the macro is that using it on the macro would result in two threads: the main thread and the thread with the current-thread runtime.

[1]: https://github.com/tokio-rs/tokio/issues/2720#issuecomment-702310198

Reply:
👍 works for me